cookiecutter/cli.py

```diff
@@ -53,7 +52,9 @@ def print_version(context, param, value):
     '-v', '--verbose',
     is_flag=True, help='Print debug information', default=False
 )
-def main(template, no_input, checkout, verbose):
+@click.argument('template')
+@click.argument('extra_context', nargs=-1)
+def main(no_input, checkout, verbose, template, extra_context):
```

Comment:
Re: POSIX; see guideline 9 at http://pubs.opengroup.org/onlinepubs/009696899/basedefs/xbd_chap12.html:

> utility_name[-a][-b][-c option_argument]
>    [-d|-e][-foption_argument][operand...]
>
> Guideline 9:
> All options should precede operands on the command line.

Now I don't think click cares about how you order things as long as the arguments to the function have the same ordering as the decorators -- changing the ordering was, AFAIK, purely aesthetic just so the code reads a little more like the way it's used.